src/test_fixture/metrics.rs

```diff
@@ -0,0 +1,49 @@
+use crate::telemetry::metrics::{INDEXED_PRSS_GENERATED, RECORDS_SENT, SEQUENTIAL_PRSS_GENERATED};
+use metrics_runtime::data::Snapshot;
+use metrics_runtime::Measurement;
+use std::collections::{BTreeMap, HashMap};
+
+/// TODO: this function is horrible and I want to fix it
+pub fn print_metrics(snapshot: Snapshot) {
+    let snapshot = snapshot.into_measurements();
+    let mut metrics_table = BTreeMap::<_, HashMap<&'static str, u64>>::new();
+    for (key, measurement) in snapshot {
+        let Measurement::Counter(metric_value) = measurement else {
+            continue
+        };
```

Comment:
I have never seen this allocation form before.  I don't think that I like it very much, but maybe it will just take some getting used to.